src/core/hiwire.c

```diff
@@ -143,6 +143,110 @@ EM_JS_NUM(int, hiwire_init, (), {
     return (!!obj) && typeof obj.then === 'function';
     // clang-format on
   };
+
+  /**
+   * Determine type and endianness of data from format. This is a helper
+   * function for converting buffers from Python to Javascript, used in
+   * PyProxyBufferMethods and in `toJs` on a buffer.
+   *
+   * To understand this function it will be helpful to look at the tables here:
+   * https://docs.python.org/3/library/struct.html#format-strings
+   *
+   * @arg format {String} A Python format string (caller must convert it to a
+   *      Javascript string).
+   * @arg errorMessage {String} Extra stuff to append to an error message if
+   *      thrown. Should be a complete sentence.
+   * @returns A pair, an appropriate TypedArray constructor and a boolean which
+   *      is true if the format suggests a big endian array.
+   * @private
+   */
+  Module.processBufferFormatString = function(formatStr, errorMessage = "")
```

Comment:
It is used in both pyproxy.js and in the update to `toJs` in #1376 which is why I separated it out to here. I will address these comments in #1411 and then merge here.

After merging more of the existing PRs, I can apply the same treatment to `hiwire_init` that I did to `pyproxy_init` (namely split into a separate js file with some macro hackery). Probably it would be good to do the same thing to `js2python` which is also mostly just one a giant `EM_JS`.

Reply:
Sounds good, thanks!

Reply:
Actually I don't really want to split out hiwire_init because it is short and it uses `#ifdef` anyways which will break javascript tooling. So I'm not sure where the most appropriate home for this function is...

Reply:
OK, to keep it here then.

Reply:
I think it's better to put this in `js2python.js` because `js2python.js` is shorter than `pyproxy.js`.

Reply:
Anyways I will leave it here for now and then move it as a separate PR after the current PRs are merged.

Reply:
Okay I moved this to `python2js_buffer.js` in #1376.